src/pages/voting/proposal/[id].tsx

```diff
@@ -1,3 +1,56 @@
-import Proposal from '../../../views/Voting/Proposal'
+import React from 'react'
+// eslint-disable-next-line camelcase
+import { SWRConfig, unstable_serialize } from 'swr'
+import { GetStaticPaths, GetStaticProps, InferGetStaticPropsType } from 'next'
+import { getAllVotes, getProposal } from 'state/voting/helpers'
+import ProposalPageRouter from 'views/Voting/Proposal'
 
-export default Proposal
+const ProposalPage = ({ fallback }: InferGetStaticPropsType<typeof getStaticProps>) => {
+  return (
+    <SWRConfig
+      value={{
+        fallback,
+      }}
+    >
+      <ProposalPageRouter />
+    </SWRConfig>
+  )
+}
+
+export const getStaticPaths: GetStaticPaths = () => {
+  return {
+    fallback: true,
+    paths: [],
+  }
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const { id } = params
+  if (typeof id !== 'string') {
+    return {
+      notFound: true,
+    }
+  }
+
+  const fetchedProposal = await getProposal(id)
+  if (!fetchedProposal) {
+    return {
+      notFound: true,
+      revalidate: 1,
+    }
+  }
+
+  const votes = await getAllVotes(fetchedProposal.id, Number(fetchedProposal.snapshot))
```

Comment:
This `getAllVotes` is quite slow, when no cache available, the routing will stuck until it is finished. 🤔
https://github.com/vercel/next.js/issues/13751

We might need to sacrifice this to only load on client right now. I can think of a possible solution is save the closed proposal voting to ipfs or redis. Ongoing proposal voting should all fetch on client.

Reply:
Done